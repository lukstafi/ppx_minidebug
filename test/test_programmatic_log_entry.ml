let db_file = "test_programmatic_log_entry.db"

let () =
  let _get_local_debug_runtime =
    let rt = Minidebug_db.debug_db_file db_file in
    fun () -> rt
  in
  let%diagn_show _logging_logic : unit =
    let rec loop logs =
      match logs with
      | "start" :: header :: tl ->
          let more =
            [%log_entry
              header;
              loop tl]
          in
          loop more
      | "end" :: tl -> tl
      | msg :: tl ->
          [%log msg];
          loop tl
      | [] -> []
    in
    ignore
    @@ loop
         [
           "preamble";
           "start";
           "header 1";
           "log 1";
           "start";
           "nested header";
           "log 2";
           "end";
           "log 3";
           "end";
           "start";
           "header 2";
           "log 4";
           "end";
           "postscript";
         ]
  in
  let db = Minidebug_client.Client.open_db db_file in
  let latest_run = Minidebug_client.Client.get_latest_run db |> Option.get in
  Minidebug_client.Client.show_trace db latest_run.run_id
