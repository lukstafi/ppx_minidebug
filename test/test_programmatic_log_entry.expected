[diagn] _logging_logic @ test/test_programmatic_log_entry.ml:8:17-8:31
  "preamble"
  [diagn] header 1 @ :0:0-0:0
    "log 1"
    [diagn] nested header @ :0:0-0:0
      "log 2"
    "log 3"
  [diagn] header 2 @ :0:0-0:0
    "log 4"
  "postscript"
